[feature] 가게 검색에 페이지네이션을 적용

jcw1031 opened the pull request:
## #️⃣ 연관된 이슈

#116 

## 📝 작업 내용

- Elasticsearch에서 검색 시 최대 250개 반환
- 캐시 서비스로부터 가져온 리스트에서 사용자가 요청한 페이지에 맞게 Stream을 사용해 페이지네이션

- 테크 스펙 : (테크 스펙 url)

### 스크린샷 (선택)

## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

---
### RCA 룰
| 코드 리뷰 멘토가 PR작성자에게 어떤 의도로 전달 되길 바라는지 알파벳으로 표현해주세요!
- r(request change): 꼭 반영
- c(comment): 왠만하면 반영
- a(approve): 반영해도 좋고 넘어가도 좋은 의견

jcw1031 committed:
[fix] 인덱스 초기화 로직 수정

- 인덱스가 생성되어있으면 초기화가 진행되지 않는 문제 해결
- 인덱스가 존재하지만 비어있다면 초기화를 진행

jcw1031 committed:
[refactor] Elasticsearch에 검색 시 최대 250개까지 ID 리스트 반환

- 캐시를 활용하고 여러 정렬 조건을 적용해 페이지네이션을 사용하기 위해 변경
- 250개라는 크기는 임의로 선정(한 페이지에 25개, 10페이지 고려)

jcw1031 committed:
[config] dev 프로필 프로퍼티 추가

- application.yml에 dev 프로필에 elasticsearch, redis, external-api 프로퍼티 추가 및 로그 제거

jcw1031 committed:
[config] RedisConnectionFactory 빈 등록 제거

- 기본으로 Bean 등록되는 팩토리로 사용
- application.yml에 설정한 프로퍼티가 사용될 수 있도록 위함

jcw1031 committed:
[feature] 검색 페이지네이션 구현

- 기본 페이지당 요소 크기 25로 설정
- Stream을 활용해 페이지네이션 구현
- 정렬 옵션은 고민해 보고 추가

jcw1031 committed:
Merge branch 'dev' into feature/116-search-pagination

hellomatia reviewed (commented):
고생많으셨습니다!!! 👍  지금 한번 확인해보았는데, 0페이지는 잘되는데, 1페이지부터 응답이 null로 오더라고요! 혹시 초기화 해주어야할 작업이 있나요?

jcw1031 commented:
> 고생많으셨습니다!!! 👍  지금 한번 확인해보았는데, 0페이지는 잘되는데, 1페이지부터 응답이 null로 오더라고요! 혹시 초기화 해주어야할 작업이 있나요?

혹시 레디스 초기화하고 해보셨나요??

jcw1031 commented:
> 고생많으셨습니다!!! 👍  지금 한번 확인해보았는데, 0페이지는 잘되는데, 1페이지부터 응답이 null로 오더라고요! 혹시 초기화 해주어야할 작업이 있나요?

아니면 희승님이 올리신 거 머지하면서 문제가 생겼을 수도 있겠네요..! 좀 이따 집 들어가면 확인해보겠습니다..!

hellomatia commented:
> 혹시 레디스 초기화하고 해보셨나요??

넵 레디스 초기화 했습니다!

jcw1031 commented:
저는 잘 출력되는데, 혹시 외부 서버를 실행하고 확인해 보셨나요!

jcw1031 committed:
[refactor] 페이지네이션 처리를 좀 더 앞당겨 리소스를 덜 사용하도록 수정

- 기존에 모든 목록을 다 조회해 ExposureRestaurantSummary로 변경한 후에 페이지네이션이 적용됨
- 미리 페이지네이션을 적용해 사이즈를 줄인 후 ExposureRestaurantSummary로 변경하도록 수정

hellomatia commented:
> 저는 잘 출력되는데, 혹시 외부 서버를 실행하고 확인해 보셨나요!

외부 서버 실행했습니다!! 엘라스틱 한번 완전히 삭제하고 다시 돌려보겠습니다!

jcw1031 commented:
> > 저는 잘 출력되는데, 혹시 외부 서버를 실행하고 확인해 보셨나요!
> 
> 외부 서버 실행했습니다!! 엘라스틱 한번 완전히 삭제하고 다시 돌려보겠습니다!

엘라스틱서치를 사용하는 서비스에서는 ID 리스트만 반환하기 때문에 아마 그 문제는 아닐 거예요..!

hellomatia commented:
```
Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2024-08-24T23:01:15.927+09:00 ERROR 78175 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'restaurantMenusRepository' defined in file [/Users/woowatech10/Team4-BF2/domain/search-domain-es/build/classes/java/main/woowa/team4/bff/search/repository/RestaurantMenusRepository.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'restaurantMenusDocumentRepository' defined in woowa.team4.bff.search.repository.RestaurantMenusDocumentRepository defined in @EnableElasticsearchRepositories declared on ElasticsearchRepositoriesRegistrar.EnableElasticsearchRepositoriesConfiguration: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:795) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:237) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1375) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1212) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:971) ~[spring-context-6.1.11.jar:6.1.11]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:625) ~[spring-context-6.1.11.jar:6.1.11]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.3.2.jar:3.3.2]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.2.jar:3.3.2]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.2.jar:3.3.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.2.jar:3.3.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.2.jar:3.3.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.2.jar:3.3.2]
	at woowa.team4.bff.RestaurantExposureServiceApplication.main(RestaurantExposureServiceApplication.java:10) ~[main/:na]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'restaurantMenusDocumentRepository' defined in woowa.team4.bff.search.repository.RestaurantMenusDocumentRepository defined in @EnableElasticsearchRepositories declared on ElasticsearchRepositoriesRegistrar.EnableElasticsearchRepositoriesConfiguration: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1806) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:600) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1443) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1353) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:904) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:782) ~[spring-beans-6.1.11.jar:6.1.11]
	... 19 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:221) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.data.repository.core.support.RepositoryFactorySupport.lambda$instantiateClass$5(RepositoryFactorySupport.java:571) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at java.base/java.util.Optional.map(Optional.java:260) ~[na:na]
	at org.springframework.data.repository.core.support.RepositoryFactorySupport.instantiateClass(RepositoryFactorySupport.java:571) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.repository.core.support.RepositoryFactorySupport.getTargetRepositoryViaReflection(RepositoryFactorySupport.java:536) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.elasticsearch.repository.support.ElasticsearchRepositoryFactory.getTargetRepository(ElasticsearchRepositoryFactory.java:80) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.data.repository.core.support.RepositoryFactorySupport.getRepository(RepositoryFactorySupport.java:317) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.lambda$afterPropertiesSet$5(RepositoryFactoryBeanSupport.java:286) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.util.Lazy.getNullable(Lazy.java:135) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.util.Lazy.get(Lazy.java:113) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.afterPropertiesSet(RepositoryFactoryBeanSupport.java:292) ~[spring-data-commons-3.3.2.jar:3.3.2]
	at org.springframework.data.elasticsearch.repository.support.ElasticsearchRepositoryFactoryBean.afterPropertiesSet(ElasticsearchRepositoryFactoryBean.java:69) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1853) ~[spring-beans-6.1.11.jar:6.1.11]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1802) ~[spring-beans-6.1.11.jar:6.1.11]
	... 30 common frames omitted
Caused by: org.springframework.data.elasticsearch.ResourceFailureException: Could not load resource from elasticsearch-settings.json
	at org.springframework.data.elasticsearch.core.ResourceUtil.readFileFromClasspath(ResourceUtil.java:48) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createSettings(IndicesTemplate.java:256) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createSettings(IndicesTemplate.java:245) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createWithMapping(IndicesTemplate.java:135) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.createIndexAndMappingIfNeeded(SimpleElasticsearchRepository.java:94) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.<init>(SimpleElasticsearchRepository.java:87) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:208) ~[spring-beans-6.1.11.jar:6.1.11]
	... 43 common frames omitted
Caused by: java.io.FileNotFoundException: class path resource [elasticsearch-settings.json] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:215) ~[spring-core-6.1.11.jar:6.1.11]
	at org.springframework.data.elasticsearch.core.ResourceUtil.readFileFromClasspath(ResourceUtil.java:45) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
	... 52 common frames omitted
```

삭제하고 다시 실행했더니 이런 오류가 계속 나타나네요

jcw1031 commented:
> ```
> Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
> 2024-08-24T23:01:15.927+09:00 ERROR 78175 --- [           main] o.s.boot.SpringApplication               : Application run failed
> 
> org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'restaurantMenusRepository' defined in file [/Users/woowatech10/Team4-BF2/domain/search-domain-es/build/classes/java/main/woowa/team4/bff/search/repository/RestaurantMenusRepository.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'restaurantMenusDocumentRepository' defined in woowa.team4.bff.search.repository.RestaurantMenusDocumentRepository defined in @EnableElasticsearchRepositories declared on ElasticsearchRepositoriesRegistrar.EnableElasticsearchRepositoriesConfiguration: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception
> 	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:795) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:237) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1375) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1212) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:971) ~[spring-context-6.1.11.jar:6.1.11]
> 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:625) ~[spring-context-6.1.11.jar:6.1.11]
> 	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.3.2.jar:3.3.2]
> 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.2.jar:3.3.2]
> 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.2.jar:3.3.2]
> 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.2.jar:3.3.2]
> 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.2.jar:3.3.2]
> 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.2.jar:3.3.2]
> 	at woowa.team4.bff.RestaurantExposureServiceApplication.main(RestaurantExposureServiceApplication.java:10) ~[main/:na]
> Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'restaurantMenusDocumentRepository' defined in woowa.team4.bff.search.repository.RestaurantMenusDocumentRepository defined in @EnableElasticsearchRepositories declared on ElasticsearchRepositoriesRegistrar.EnableElasticsearchRepositoriesConfiguration: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1806) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:600) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1443) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1353) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:904) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:782) ~[spring-beans-6.1.11.jar:6.1.11]
> 	... 19 common frames omitted
> Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception
> 	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:221) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.data.repository.core.support.RepositoryFactorySupport.lambda$instantiateClass$5(RepositoryFactorySupport.java:571) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at java.base/java.util.Optional.map(Optional.java:260) ~[na:na]
> 	at org.springframework.data.repository.core.support.RepositoryFactorySupport.instantiateClass(RepositoryFactorySupport.java:571) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.repository.core.support.RepositoryFactorySupport.getTargetRepositoryViaReflection(RepositoryFactorySupport.java:536) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.elasticsearch.repository.support.ElasticsearchRepositoryFactory.getTargetRepository(ElasticsearchRepositoryFactory.java:80) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.data.repository.core.support.RepositoryFactorySupport.getRepository(RepositoryFactorySupport.java:317) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.lambda$afterPropertiesSet$5(RepositoryFactoryBeanSupport.java:286) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.util.Lazy.getNullable(Lazy.java:135) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.util.Lazy.get(Lazy.java:113) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.afterPropertiesSet(RepositoryFactoryBeanSupport.java:292) ~[spring-data-commons-3.3.2.jar:3.3.2]
> 	at org.springframework.data.elasticsearch.repository.support.ElasticsearchRepositoryFactoryBean.afterPropertiesSet(ElasticsearchRepositoryFactoryBean.java:69) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1853) ~[spring-beans-6.1.11.jar:6.1.11]
> 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1802) ~[spring-beans-6.1.11.jar:6.1.11]
> 	... 30 common frames omitted
> Caused by: org.springframework.data.elasticsearch.ResourceFailureException: Could not load resource from elasticsearch-settings.json
> 	at org.springframework.data.elasticsearch.core.ResourceUtil.readFileFromClasspath(ResourceUtil.java:48) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createSettings(IndicesTemplate.java:256) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createSettings(IndicesTemplate.java:245) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createWithMapping(IndicesTemplate.java:135) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.createIndexAndMappingIfNeeded(SimpleElasticsearchRepository.java:94) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.<init>(SimpleElasticsearchRepository.java:87) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[na:na]
> 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[na:na]
> 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[na:na]
> 	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:208) ~[spring-beans-6.1.11.jar:6.1.11]
> 	... 43 common frames omitted
> Caused by: java.io.FileNotFoundException: class path resource [elasticsearch-settings.json] cannot be opened because it does not exist
> 	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:215) ~[spring-core-6.1.11.jar:6.1.11]
> 	at org.springframework.data.elasticsearch.core.ResourceUtil.readFileFromClasspath(ResourceUtil.java:45) ~[spring-data-elasticsearch-5.3.2.jar:5.3.2]
> 	... 52 common frames omitted
> ```
> 
> 삭제하고 다시 실행했더니 이런 오류가 계속 나타나네요

오잉.. 왜 저는 삭제하고 해도 잘 될까요..ㅜ

hellomatia commented:
확인해보니 희승님이 서버 분리를 하면서 csv파일들이 다 사라졌는데, 이 파일들은 도커파일에 추가되는건가요?

jcw1031 commented:
> 확인해보니 희승님이 서버 분리를 하면서 csv파일들이 다 사라졌는데, 이 파일들은 도커파일에 추가되는건가요?

아앗 제가 볼륨을 삭제 안 하고 테스트 했었네요..! 확인해 보니까 그렇네용
아래 노션 링크로 들어가서 csv 파일 두 개 다운받으시고 `domain:search-domain-es` 모듈의 `src/main/resources` 경로에 추가하고 실행하시면 됩니다!

[Elasticsearch 초기화 데이터](https://www.notion.so/woopaca/Docker-b6e2d3230de64157b8a38b641ea06240?pvs=4)

huiseung commented:
> 확인해보니 희승님이 서버 분리를 하면서 csv파일들이 다 사라졌는데, 이 파일들은 도커파일에 추가되는건가요?

원래 csv 파일은 github에 올라가지 않고 slack/notion으로 공유되도록 하고 있어요.

예전 gitignore 참조
mono/src/main/resources/menu_es_1.csv
mono/src/main/resources/restaurant_es_1.csv

hellomatia commented:
처음부터 다 초기화 하니까 잘 작동합니다!! 👍 고생하셨습니다!!

jcw1031 commented:
> > 확인해보니 희승님이 서버 분리를 하면서 csv파일들이 다 사라졌는데, 이 파일들은 도커파일에 추가되는건가요?
> 
> 원래 csv 파일은 github에 올라가지 않고 slack/notion으로 공유되도록 하고 있어요.
> 
> 예전 gitignore 참조 mono/src/main/resources/menu_es_1.csv mono/src/main/resources/restaurant_es_1.csv

도커 실행 시 초기화될 수 있도록 변경해 보겠습니다!

jcw1031 commented:
> 처음부터 다 초기화 하니까 잘 작동합니다!! 👍 고생하셨습니다!!

넵 ㅎㅎ 고생하셨습니다!
